velox/functions/sparksql/ArrayRepeat.cpp

```diff
@@ -0,0 +1,196 @@
+    }
+
+    auto count = constantCount->valueAt(0);
+    if (count < 0) {
+      // Negative count is like zero count, and empty array will be returned.
+      count = 0;
+    } else if (count > kMaxCount) {
+      // The max limit is aligned with Spark.
+      auto pex = std::make_exception_ptr(std::invalid_argument(fmt::format(
+          "({} vs. {}) Count argument of array_repeat function must be less than or equal to (MAX_INT32 - 15)",
+          count,
+          kMaxCount)));
+      context.setErrors(rows, pex);
+      return nullptr;
+    }
+
+    const auto totalCount = count * numRows;
+
+    // Allocate new vectors for indices, lengths and offsets.
+    BufferPtr indices = allocateIndices(totalCount, pool);
+    BufferPtr sizes = allocateSizes(numRows, pool);
+    BufferPtr offsets = allocateOffsets(numRows, pool);
+    auto rawIndices = indices->asMutable<vector_size_t>();
+    auto rawSizes = sizes->asMutable<vector_size_t>();
+    auto rawOffsets = offsets->asMutable<vector_size_t>();
+
+    vector_size_t offset = 0;
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      rawSizes[row] = count;
+      rawOffsets[row] = offset;
+      std::fill(rawIndices + offset, rawIndices + offset + count, row);
+      offset += count;
+    });
+
+    return std::make_shared<ArrayVector>(
+        pool,
+        outputType,
+        nullptr,
+        numRows,
+        offsets,
+        sizes,
+        BaseVector::wrapInDictionary(nullptr, indices, totalCount, args[0]));
+  }
+
+  VectorPtr applyNonConstant(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context) const {
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    auto countDecoded = decodedArgs.at(1);
+    int32_t totalCount = 0;
+    // Directly throws user exception if invalid count is used.
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      if (countDecoded->mayHaveNulls() && countDecoded->isNullAt(row)) {
+        return;
+      }
+      auto count = countDecoded->valueAt<int32_t>(row);
+      if (count < 0) {
+        // Negative count is like zero count, and empty array will be returned.
+        count = 0;
+      } else {
+        // The max limit is aligned with Spark.
+        VELOX_USER_CHECK_LE(
+            count,
+            kMaxCount,
+            "Count argument of array_repeat function must be less than or equal to (MAX_INT32 - 15)");
+      }
+      totalCount += count;
+    });
+
+    const auto numRows = rows.end();
+    auto pool = context.pool();
+
+    // Allocate new vector for nulls if necessary.
+    BufferPtr nulls;
+    uint64_t* rawNulls = nullptr;
+    if (countDecoded->mayHaveNulls()) {
+      nulls = allocateNulls(numRows, pool);
+      rawNulls = nulls->asMutable<uint64_t>();
+    }
+
+    // Allocate new vectors for indices, lengths and offsets.
+    BufferPtr indices = allocateIndices(totalCount, pool);
```

Comment:
What if totalCount is zero?

Reply:
If totalCount is zero, empty array is the result for each row. Just added a test for this case. Thanks!